### PR TITLE
Fixes bsa artillery construction issues.

### LINKED ISSUE #23010: You  can  create free infinite glass and wires using the bluespace artillery computer board because the computer board only goes into machines but makes a computer
Reproduce:
Step 1: Be oranges and fail to read my PR reporting  a infinite glass bug.

Step 1b: Put the computer board in a machine frame because it only goes in machines(for some reason) screwdriver it.

Step 2:  Deconstruct with screwdriver it turns into a computer. Decon from  there.

Also glitch 2 or fuckup or whatever. Once the parts are built you can't deconstruct them even though they have to be built in A HIGHLY SPECIFIC ORDER. These things cannot be screwdrivered and deconned nor can they be wrenched to be moved

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #23010
If anyone feels like writing an ic guide, i'll include it in the part crate.

### CODE REVIEW

File: code/modules/station_goals/bsa.dm [hunk 1]
```diff
@@ -75,12 +81,17 @@
...
 		return ..()
 
 /obj/machinery/bsa/middle/proc/check_completion()
 	if(!front || !back)
 		return "No linked parts detected!"
+	if(!front.anchored || !back.anchored || !anchored)
+		return "Linked parts unwrenched!"
 	if(front.y != y || back.y != y || !(front.x > x && back.x < x || front.x < x && back.x > x) || front.z != z || back.z != z)
 		return "Parts misaligned!"
 	if(!has_space())
```
Comment (coiax) [inline code/modules/station_goals/bsa.dm]: These messages are never passed to the user.

Comment (AnturK) [inline code/modules/station_goals/bsa.dm]: They will now, ui_data runtimed on null read.

**Comment by Kromgar** [comment]
Congatulations on your ownership of your new [Bluespace Artillery Cannon]. To construct this fine piece of Nanotrasen Weaponry you are going to need X things

1. 20 bluespace crystals.
2. Quadratic Capatiors
3. A 10 foot open space either east or west of the bluespace artillery
4. A screwdriver


To properly construct your Bluespace Artillery Cannon you must know that it can't be built facing north or south it can only be built facing west or east. The components must be built in a specific order and with enough space or the machine will not unfold from bluespace.

1. The artillery generator is the rear component.
2. Fusor is center component.
3.  The Bore if the front of the cannon.
4. Computer can be anywhere in the same room

If the bore and fusor are facing west from the generator the cannon will unfold facing west and vice versa.

Once the components are constructed you must link them to the fusor and then activate the deployment process to unfold the cannon. Once unfolded if there is enough power to the room's APC the cannon can be fired indefinitely. 

To Fire: Simply select gps coordinates and then press the fire button. The Bluespace Artillery Cannon has a charging time. Do not stare down the barrel while firing

**Comment by coiax** [comment]
Also, the barrel needs to be pointing to space, although if we're honest, there's no way to fire this thing that isn't ultra destructive. The explosion it makes is quite small, what is FAR more damaging is the ex_act(1) of all turfs in the way

**Comment by Kromgar** [comment]
I gotta say the BSA should really be made more powerful for all the work you put into the damn thing.

But i didn't put it in the manual because it should be "common sense" and its more fun if people fuck up

**Comment by optimumtact** [comment]
Hah your guide will never get in now!!!!!!!!!!!

**Pull request merged**